File: Java/src/main/java/com/easytoolsoft/concurrentprogramming/StartUp.java
package com.easytoolsoft.concurrentprogramming;

import com.easytoolsoft.concurrentprogramming.cli.CliCommand;

public class StartUp {
	public static void main(final String[] args) {
		if (args == null || args.length == 0) {
			return;
		}

		String className = String.format("com.easytoolsoft.concurrentprogramming.cli.%sCliCommand", args[0]);
		try {
			CliCommand cliCommand = (CliCommand) Class.forName(className).newInstance();
			cliCommand.execute(args);
		} catch (InstantiationException | IllegalAccessException | ClassNotFoundException e) {
			e.printStackTrace();
		}
	}
}

File: Scala/src/main/scala/com/easytoolsoft/concurrentprogramming/StartUp.scala
package com.easytoolsoft.concurrentprogramming

import com.easytoolsoft.concurrentprogramming.cli.CliCommand

object StartUp {
  def main(args: Array[String]): Unit = {
    if (args.length > 0) {
      try {
        val className = String.format("com.easytoolsoft.concurrentprogramming.cli.%sCliCommand", args(0))
        val cliCommand = Class.forName(className).newInstance().asInstanceOf[CliCommand]
        cliCommand.execute(args)
      } catch {
        case t: Throwable => t.printStackTrace()
      }
    }
  }
}
